ambari-server/src/main/java/org/apache/ambari/server/security/encryption/Encryptor.java

```diff
@@ -36,8 +35,20 @@
    * 
    * @param decryptible
    *          to be decrypted
-   * @return the decrypted value
    */
   void decryptSensitiveData(T decryptible);
 
+  /**
+   * @return the default encryption key used by this encryptor
+   */
+  String getEncryptionKey();
+
+  Encryptor NONE = new Encryptor() {
+    @Override
+    public void encryptSensitiveData(Object data) { }
+    @Override
+    public void decryptSensitiveData(Object decryptible) { }
+    @Override
+    public String getEncryptionKey() { return ""; }
```

Comment:
how about using null instead of empty string, so we get an NPE if this key is attempted to really be used somewhere?

Reply:
it's intentionally an empty string so when encryption is not enabled the salt string will be empty